compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaPathParameterSymbol.java

```diff
@@ -123,12 +141,13 @@ public String signature() {
 
         String typeSignature;
         if (this.pathSegmentKind() == Kind.PATH_REST_PARAMETER) {
-            typeSignature = ((ArrayTypeSymbol) this.typeDescriptor()).memberTypeDescriptor().signature() + "...";
+            typeSignature = this.typeDescriptor().signature() + "...";
         } else {
             typeSignature = this.typeDescriptor().signature();
         }
 
-        this.signature = "[" + typeSignature + " " + this.getName().get() + "]";
+        String paramName = this.getName().orElse("");
+        this.signature = "[" + typeSignature + " " + paramName + "]";
```

Comment:
Is this correct if there is no param name?

Reply:
Yes, the signature should show something like this `[string]` or [string...]`

Reply:
Doesn't this add empty space at the end?